dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java

```diff
@@ -193,7 +193,36 @@ public HpccRemoteFileReader(DataPartition dp, FieldDef originalRD, IRecordBuilde
      * @throws Exception
      * 			  general exception
      */
-    public HpccRemoteFileReader(DataPartition dp, FieldDef originalRD, IRecordBuilder recBuilder, int connectTimeout, int limit, boolean createPrefetchThread, int readSizeKB, FileReadResumeInfo resumeInfo, int socketOpTimeoutMs) throws Exception
+    public HpccRemoteFileReader(DataPartition dp, FieldDef originalRD, IRecordBuilder recBuilder, int connectTimeout, int limit, boolean createPrefetchThread, int readSizeKB, FileReadResumeInfo resumeInfo, int socketOpTimeoutMs) throws Exception {
+        this(dp, originalRD, recBuilder, connectTimeout, limit, createPrefetchThread, readSizeKB, resumeInfo, RowServiceInputStream.DEFAULT_SOCKET_OP_TIMEOUT_MS,null);
+    }
+    /**
+     * A remote file reader that reads the part identified by the HpccPart object using the record definition provided.
+     *
+     * @param dp
+     *            the part of the file, name and location
+     * @param originalRD
+     *            the record defintion for the dataset
+     * @param recBuilder
+     *            the IRecordBuilder used to construct records
+     * @param connectTimeout
+     *            the connection timeout in milliseconds, -1 for default
+     * @param limit
+     *            the maximum number of records to read from the provided data partition, -1 specifies no limit
+     * @param createPrefetchThread
+     *            the input stream should create and manage prefetching on its own thread. If false prefetch needs to be called on another thread periodically.
+     * @param readSizeKB
+     *            read request size in KB, -1 specifies use default value
+     * @param resumeInfo
+     *            FileReadeResumeInfo data required to restart a read from a particular point in a file, null for reading from start
+     * @param socketOpTimeoutMs
+     *            Socket (read / write) operation timeout in milliseconds
+     * @param fileName
+     *            filename to read
+     * @throws Exception
+     * 			  general exception
+     */
+    public HpccRemoteFileReader(DataPartition dp, FieldDef originalRD, IRecordBuilder recBuilder, int connectTimeout, int limit, boolean createPrefetchThread, int readSizeKB, FileReadResumeInfo resumeInfo, int socketOpTimeoutMs, String fileName) throws Exception
```

Comment:
Passing in the fileName here is a great idea to improve the logging output, and it probably should always have been available within DataPartition. Would you be willing to move fileName into DataPartition? Ideally we would then pass in the filename to DataPartition.createPartitions from HPCCFile.